core/src/main/java/feast/core/controller/CoreServiceRestController.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2019 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.controller;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import feast.core.grpc.CoreServiceResponseGenerator;
+import feast.proto.core.CoreServiceProto.GetFeastCoreVersionResponse;
+import feast.proto.core.CoreServiceProto.GetFeatureSetRequest;
+import feast.proto.core.CoreServiceProto.ListFeatureSetsRequest;
+import feast.proto.core.CoreServiceProto.ListFeaturesRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(produces = "application/json")
+public class CoreServiceRestController {
+
+  private final JsonFormat.Printer jsonPrinter;
+  @Autowired private final CoreServiceResponseGenerator responseGenerator;
+
+  @Autowired
+  public CoreServiceRestController(CoreServiceResponseGenerator responseGenerator) {
+    this.jsonPrinter = JsonFormat.printer();
+    this.responseGenerator = responseGenerator;
+  }
+
+  @RequestMapping(value = "/api/v1/version", method = RequestMethod.GET)
+  @ResponseBody
+  public String getVersion() throws InvalidProtocolBufferException {
+    GetFeastCoreVersionResponse response = responseGenerator.generateGetFeastCoreVersionResponse();
+    return jsonPrinter.print(response);
+  }
+
+  @RequestMapping(
+      value = "/api/v1/projects/{project}/feature-sets/{featureSet}",
```

Comment:
Nitpick. Lets reserve plurals for when talking about mutliple `projects`  or `feature-sets`. Use `project` and `feature-set` respectively.

Reply:
@mrzzy how would we list all feature-sets? Wouldn't that be at `/api/v1/projects/{project}/feature-sets`?

If we only look at feature-sets then it would be weird to have `/api/v1/projects/{project}/feature-set` return multiple feature sets, so I think that is less bad than having `/api/v1/projects/{project}/feature-sets/123` point to a single feature set.

I dont think either option is wrong though.

Possibly another question that we have to ask is whether `/projects/{project}` has to be there. I am not sure if we will ever list `/projects` out, although it could theoretically happen. Raises the question of whether we should follow the Google style query parameter for projects instead.

Reply:
Hmm, the code for this diff is gone. I think what I was referring to 
> /api/v1/projects/{project}/feature-sets/123

as being unsavory uses purals.

I am OK with what is currently implemented. 
https://github.com/feast-dev/feast/blob/f666287128f7cb3eed8406d83a0494fe5653f718/core/src/main/java/feast/core/controller/CoreServiceRestController.java#L106

Although its more technically tricky this is what I favor IMO:
```
/api/v1/feature-set/{feature-set-ref} 
```
or we can replace with the Feature Set's auto generated id
```
/api/v1/feature-set/42
```

Reply:
This endpoint will be removed as `/feature-sets` can do the same and more.